crates/geometry/src/arc.rs

```diff
@@ -46,23 +53,23 @@ impl RelativeEq for Arc {
     }
 }
 
-impl Arc {
-    pub fn new(circle: Circle, start: Point2<f32>, end: Point2<f32>) -> Self {
+impl<Frame> Arc<Frame> {
+    pub fn new(circle: Circle<Frame>, start: Point2<Frame>, end: Point2<Frame>) -> Self {
         Self { circle, start, end }
     }
 
-    pub fn length(&self, orientation: Orientation) -> f32 {
+    pub fn length(&self, orientation: Direction) -> f32 {
         let vector_start = self.start - self.circle.center;
         let vector_end = self.end - self.circle.center;
 
-        let angle_x_axis_to_start = vector_start.y.atan2(vector_start.x);
-        let mut angle = vector_end.y.atan2(vector_end.x) - angle_x_axis_to_start;
+        let angle_x_axis_to_start = vector_start.y().atan2(vector_start.x());
+        let mut angle = vector_end.y().atan2(vector_end.x()) - angle_x_axis_to_start;
```

Comment:
you said something about replacing atan?

Reply:
There are two ways to calculate angles between vectors:
`Vector::angle` always returns the smaller angle between the two vectors and the angle is unsigned.
`UnitComplex::rotation_between` returns a rotation that can transform one vector to the other and is effectively just a wrapper around atan2.
I don't particularly like the idea of doing `UnitComplex::rotation_between().angle()`, but perhaps we can add a yet unnamed function to our vector type which just returns the signed angle.

What do you think?